var/spack/repos/spack_repo/builtin/packages/grid/package.py

```diff
@@ -68,12 +100,96 @@ class Grid(AutotoolsPackage):
 
     depends_on("doxygen", type="build", when="+doxygen-doc")
 
+    conflicts(
+        "+cuda",
+        when="+rocm",
+        msg="CUDA / ROCm are mututally exclusive. At most 1 GPU platform can be configured",
+    )
+    conflicts("+cuda", when="platform=darwin", msg="There is no GPU support for macOS")
+    conflicts(
+        "cuda_arch=none",
+        when="+cuda",
+        msg="Must specify CUDA compute capabilities of your GPU, see https://developer.nvidia.com/cuda-gpus",
+    )
+    conflicts(
+        "+accelerator-aware-mpi",
+        when="-cuda -rocm",
+        msg="Cannot compile for GPU-aware MPI when not compiling for GPU",
+    )
+    conflicts(
+        "shared-memory=nvlink",
+        when="-cuda -rocm",
+        msg="Cannot compile with nvlink when not compiling for GPU",
+    )
+
     def autoreconf(self, spec, prefix):
         Executable("./bootstrap.sh")()
 
+    def setup_build_environment(self, env):
+        spec = self.spec
+        env.set("CXXFLAGS", self.compiler.cxx17_flag)
+
+        if spec.satisfies("+cuda") or spec.satisfies("+rocm"):
+            if spec.satisfies("+cuda"):
```

Comment:
Why double nesting and not just:
```python
if spec.satisfies("+cuda"):
    ...
elif spec.satisfies("+rocm"):
    ...
else:
    ...
```
?

Reply:
Can't remember the motivation for that, so I just simplified it, thanks!